### PR TITLE
update to pull in VM trait from external location

### PR DESCRIPTION
**Pull request by alexytsu**
The other side of: https://github.com/filecoin-project/builtin-actors/issues/1327

- pulls in the trait, associated types and utils from external location
- deletes parallel/duplicated code
- general dependency cleanup

TODOs
- [x] Implement Actor Manifest in a generic way (ref-fvm and builtin-actors have incompatible representations)
- [x] Expose state root and other traits

To followup:
- [ ] Automatic conversion/derivation of execution traces and message results
- [ ] Tracking of execution traces for builtin-actors' tests to make assertions

### CODE REVIEW

File: builtin/src/genesis.rs [hunk 1]
```diff
@@ -177,7 +179,7 @@ pub fn create_genesis_actors<B: WorkbenchBuilder>(
     )?;
 
     // Faucet account
-    let faucet_id = BURNT_FUNDS_ACTOR_ID - 1;
+    let faucet_id = TEST_FAUCET_ADDR.id().unwrap();
```
Comment (alexytsu) [inline builtin/src/genesis.rs]: @anorth Changing this from `BURNT_FUNDS_ACTOR_ID - 1` (98) to `TEST_FAUCET_ADDR` (102) breaks the `create_accounts` call for some reason. 

The FVM itself sees a zero balance for the faucet_addr and the transfer during the send fails with "insufficient funds".  Even though if I call `v.actor(TEST_FAUCET_ADDR).balance` to load the state directly before the send, I see the correct balance. 

I'm not sure what is going on here. If I patch locally to builtin-actors and modify its `create_accounts` to use the ID 98 instead, it all seems to work. Is there some fundamental difference between an actor using a ID below or above FIRST_NON_SINGLETON_ID.

Comment (alexytsu) [reply builtin/src/genesis.rs]: https://github.com/anorth/fvm-workbench/pull/23 demonstrates the "fix"

Comment (anorth) [reply builtin/src/genesis.rs]: I'm sure there's a good explanation, but I don't know what it is. There's nothing fundamentally different about actor IDs <100, I don't think the VM handles them any differently. There might be hardcoded assumptions somewhere (in actors) that we're missing.

Comment (alexytsu) [reply builtin/src/genesis.rs]: InitActor was being created with a default state. So it was allocating IDs to new actors that overlapped with the test_faucet_address (and anything else seeded as a part of genesis state)

File: api/src/wrangler.rs [hunk 2]
```diff
@@ -29,6 +25,7 @@ pub struct ExecutionWrangler {
     gas_limit: u64,
     gas_fee_cap: TokenAmount,
     gas_premium: TokenAmount,
+    circulating_supply: RefCell<TokenAmount>,
```
Comment (anorth) [inline api/src/wrangler.rs]: This is an odd place for this field to be. I would expect it to be at least down in the underlying bench, if not deeper. Setting this value has to affect WASM-compiled actor code that executes on the workbench when it calls the syscall to fetch this value. So putting it here is probably wrong.

File: builtin/tests/builtin_actors_tests.rs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+use fil_actors_integration_tests::tests::withdraw_balance_success_test;
+use fvm_workbench_builtin_actors::setup;
+
+#[test]
+fn withdraw_balance_test() {
+    let w = setup();
+    withdraw_balance_success_test(&w);
```
Comment (anorth) [inline builtin/tests/builtin_actors_tests.rs:7]: 🎉

File: vm/src/bench/mod.rs [hunk 4]
```diff
@@ -74,8 +77,11 @@ where
         Ok(raw.map(|a| ActorState {
             code: a.code,
             state: a.state,
-            sequence: a.sequence,
+            call_seq: a.sequence,
             balance: a.balance,
+            // TODO: possibly rename predictable address if these are the same concept
+            // in ref-fvm predictable address is assigned to delegated address in some instances
```
Comment (anorth) [inline vm/src/bench/mod.rs:84]: These are the same concept as far as I know.

**Review by anorth — COMMENTED** [review]
Looks good except for the circulating supply on wrangler. Let's chat about it today.

File: api/src/wrangler.rs (hunk 2 above)
Comment (anorth) [reply api/src/wrangler.rs]: As discussed, this should go along with fields like epoch, base fee. In the FVM implementation these fields are in the MachineContext or NetworkConfig held by the machine.

File: vm/src/bench/mod.rs (hunk 4 above)
Comment (alexytsu) [reply vm/src/bench/mod.rs:84]: Will do in a followup: requires changing and landing in builtins-first

File: api/src/wrangler.rs (hunk 2 above)
Comment (alexytsu) [reply api/src/wrangler.rs]: I've deferred exposing NetworkConfig for now as it is a concept owned by the short-lived Kernel during execution and likely not relevant for exposing.

MachineContext has a large amount of overlap so exposing that instead should be sufficient. Like the rename predictable_address -> delegated_address. I'll make the change in vm_api (builtins repo) first and have the changes flow here in a followup.

**Review by anorth — APPROVED** [review]

**Pull request merged**